app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -462,6 +462,14 @@ def import_uploads
         upload.permit(:id, :content_unit_id, :size, :checksum, :name, :digest).to_h
       end
 
+      if @repository.content_type != 'docker' && uploads.first['checksum'].nil?
+        fail HttpErrors::BadRequest, _('Checksum is a required parameter.')
+      end
+
+      if uploads.first['name'].nil?
+        fail HttpErrors::BadRequest, _('Name is a required parameter.')
+      end
+
```

Comment:
This could go back to to being required in api description I think..

Reply:
Updated name to be a required param

Reply:
We don't need this anymore with name now being a required param. 
```
if uploads.first['name'].nil?
   fail HttpErrors::BadRequest, _('Name is a required parameter.')
end
```

Reply:
After discussing on IRC, hammer will honor the required param but rails is not when using a script to talk to the API, so keeping this validation.